### ISSUE TITLE
Autocomplete places positional arguments after keyword arguments

### ISSUE THREAD

**Issue by QuaternionsRock** [microsoft/pylance-release#4173]
<h3> Environment data </h3>

-   Language Server version: 2023.3.30
-   OS and version: win32 x64
-   Python version (and distribution if applicable, e.g. Anaconda): 
-   python.analysis.indexing: true
-   python.analysis.typeCheckingMode: off

<h3> Code Snippet </h3>

I found this bug while writing a subclass of `matplotlib.scale.LogitScale`. For context, here is the relevant snippet from the `pyi` file:

```python
class LogitScale(ScaleBase):

    name = ...
    def __init__(
        self,
        axis: Axis,
        nonpositive: Literal["mask", "clip"] = ...,
        *,
        one_half: str = ...,
        use_overline=...
    ) -> None: ...
```

I started writing the `__init__` method of my subclass:

```python
class Foo(LogitScale):
    def __
```

Then used autocomplete to fill in the rest:

```python
class Foo(LogitScale):
    def __init__(self, axis: Axis, nonpositive: Literal["mask", "clip"] = ..., *, one_half: str = ..., use_overline=...) -> None:
        super().__init__(axis, nonpositive, one_half=one_half, use_overline)
```

This completion contains a syntax error, `use_overline` is a keyword-only parameter. Additionally, it appears after `one_half` keyword argument. For clarity, this is the correct completion:

```python
class Foo(LogitScale):
    def __init__(self, axis: Axis, nonpositive: Literal["mask", "clip"] = ..., *, one_half: str = ..., use_overline=...) -> None:
        super().__init__(axis, nonpositive, one_half=one_half, use_overline=use_overline)
```

**Comment by rchiodo** [comment]
Thanks for the issue.

**Comment by erictraut** [comment]
@rchiodo, I just checked in a fix for this in the pyright repo. There was a small flaw in the logic that produces the override body.

**Comment by rchiodo** [comment]
With that in mind, this should be fixed next week when we release our next prerelease version.

**Comment by rchiodo** [comment]
This issue has been fixed in prerelease version 2023.4.11, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023411-5-April-2023-PreRelease)
